src/utils/index.js

```diff
@@ -17,7 +17,18 @@ const getGitHash = () => {
   }
 };
 
+let earlyVerboseStatements = [];
+
 const verbose = (msg, ...rest) => {
+  /**
+   * During server start-up there is a race condition where calls to `utils.verbose()`
+   * can occur before arg parsing, which means global.verbose has not been set.
+   * We store these & print them immediately following our "server loaded" message.
+   */
+  if (global.verbose===undefined) {
+    earlyVerboseStatements.push([msg, ...rest]);
+    return;
+  }
```

Comment:
_non-blocking_

This seems very hacky to me. For example, the verbose messages will still be lost if there's a crash between the verbose call and the call to `_logEarlyVerboseStatements`.

I'd rather see us move from an option to an environment variable, e.g. `NEXTSTRAIN_DEBUG=1` or similar. Env vars are commonly used to control debugging logs for this reason, i.e. that they're actually global at process start (among other reasons).

Maybe as part of a larger [logging improvement](https://github.com/nextstrain/nextstrain.org/issues/511) effort…

Reply:
This also shouldn't be needed… it used to work. I suspect the change to use ES modules affected global handling in a non-obvious way.

Reply:
I propose a different fix that restores how it used to work: https://github.com/nextstrain/nextstrain.org/pull/679

Reply:
Yeah, +1 for the larger logging improvement.

> This also shouldn't be needed… it used to work. I suspect the change to use ES modules affected global handling in a non-obvious way.

The manifest `utils.verbose()` calls used to work because they were inside a promise (and thus delayed). I'm not sure if the logging from 6dff7d23abf6f478d6ce2345c81c568992f8e86f ever worked, but it may well have done!